app/dao/inbound_sms_dao.py

```diff
@@ -62,6 +63,35 @@ def dao_count_inbound_sms_for_service(service_id, limit_days):
     ).count()
 
 
+def _insert_update_notification_history(subquery, query_limit=10000):
+    offset = 0
+    inbound_sms_query = db.session.query(
+        *[x.name for x in InboundSmsHistory.__table__.c]
+    ).filter(InboundSms.id.in_(subquery))
+    inbound_sms_count = inbound_sms_query.count()
+
+    while offset < inbound_sms_count:
+        statement = insert(InboundSmsHistory).from_select(
+            InboundSmsHistory.__table__.c,
+            inbound_sms_query.limit(query_limit).offset(offset)
+        )
+
+        statement = statement.on_conflict_do_update(
+            constraint="inbound_sms_history_pkey",
+            set_={
+                "created_at": statement.excluded.created_at,
```

Comment:
hmm wondering if we should be setting the created_at as when the original inbound sms was created or if instead we should be saving that information on a different field (does `provider_date` cover it?) and letting `created_at` refer to the creation of the `inbound_sms_history` record.

Reply:
Provider date is when we received the inbound sms from provider I think, and created at is when we created an object for it. 

So probably good call that in history table, created_at would be when we create a history object.

Reply:
Hmm, I think your original method was better - my reckon is that we're more likely to want to check an inbound sms by when it was originally created at. The nightly task is an implementation detail and having the created_at change meaning between the two tables feels like a big gotcha. It might be useful to know when the inbound sms was received if we need to align it with log messages, service api calls, etc.

in outbound-land, notification_history.created_at is a copy of notification.created_at. I think we should maintain that behaviour. You can verify that behaviour by running `select max(created_at) from notification_history;`. For example, despite the task having most recently run last night, the newest created_at time on the preview notification_history table is from the 14th.

Reply:
Ok, going with @leohemsted 's suggestion, it is also much easier to leave it that way 😁